--- a/src/system.make
+++ b/src/system.make
@@ -13,10 +13,10 @@
 COMPFLAGS_F77= $(COMPFLAGS)
 
 # Set BLAS and LAPACK libraries
-BLAS= -lvecLibFort
+BLAS= -llapack -lblas
 
 # Full library call; remove scalapack if using dummy diag module
-LIBS= $(FFT_LIB) $(XC_LIB) -lscalapack $(BLAS)
+LIBS= $(FFT_LIB) $(XC_LIB) -lscalapack-openmpi $(BLAS)
 
 # LibXC compatibility (LibXC below) or Conquest XC library
 
@@ -30,7 +30,7 @@
 #XC_LIBRARY = LibXC_v2
 XC_LIBRARY = LibXC
 XC_LIB = -lxcf90 -lxc
-XC_COMPFLAGS = -I/usr/local/include
+XC_COMPFLAGS = -I/usr/include
 
 # Set FFT library
 FFT_LIB=-lfftw3
--- a/tools/BasisGeneration/system.make
+++ b/tools/BasisGeneration/system.make
@@ -13,10 +13,10 @@
 COMPFLAGS_F77= $(COMPFLAGS)
 
 # Set BLAS and LAPACK libraries
-BLAS= -lvecLibFort
+BLAS= -llapack -lblas
 
 # Full library call; remove scalapack if using dummy diag module
-LIBS= $(FFT_LIB) $(XC_LIB) -lscalapack $(BLAS)
+LIBS= $(FFT_LIB) $(XC_LIB) -lscalapack-openmpi $(BLAS)
 
 # LibXC compatibility (LibXC below) or Conquest XC library
 
@@ -30,7 +30,7 @@
 #XC_LIBRARY = LibXC_v2
 XC_LIBRARY = LibXC
 XC_LIB = -lxcf90 -lxc
-XC_COMPFLAGS = -I/usr/local/include
+XC_COMPFLAGS = -I/usr/include
 
 # Set FFT library
 FFT_LIB=-lfftw3
--- a/src/Makefile
+++ b/src/Makefile
@@ -96,9 +96,9 @@
 	gzip ../$(TARNAME)
 
 clean:
-	rm -f *.o *.mod *~  *.d work.pc* deps.obj.inc
+	rm -f *.o *.mod *~  *.d work.pc* deps.obj.inc datestamp.f90 makedeps.txt ../bin/$(TARGET)
 #	(cd FFT; make -k clean)
-	(cd utilities; make -k clean)
+#	(cd utilities; make -k clean)
 
 clean_lib:
 	rm -f *.a
--- a/tools/BasisGeneration/Makefile
+++ b/tools/BasisGeneration/Makefile
@@ -27,3 +27,6 @@
 
 %.o: %.f90
 	$(FC) $(COMPFLAGS) -c $<
+
+clean:
+	rm -f *.o *.mod MakeIonFiles datestamp.f90 deps.obj ../../bin/$(TARGET)
